### PR TITLE
Dbatiste/refactor es6

### PR DESCRIPTION
**Pull request by dbatiste**
*Do not merge!*

@dlockhart : love to get your thoughts!  There are probably a couple more changes I would make, but thought I would throw this out there to get your input.  ex. I've completely disabled tests for the moment. :)

These changes implement a possible solution to provide localized terms in a React app using ES6 classes.  The previous solution uses React mixins, which are not supported in React when using ES6.  This solution is similar to what's discussed in [react-intl](https://github.com/yahoo/react-intl/issues/99).  This is meant to be a temporary solution until react-intl is refactored to address this issue, however this work ultimately depends on React 0.14 changes that include changing the context chain to follow parent-child relationship rather than the ownership chain.  

Additional changes will be required to provide `formatMessage` functionality to handle replacements.

The approach taken is to use a factory to construct a high-level component responsible for setting up the context, which is accessible from child components.  Setup entails wrapping the app in a component to set up the context:
```
// messages : json as we already have it
// Application : the root app component

const internationalize = require('react-frau-intl').internationalize;

let IntlApplication = internationalize(Application);
React.render(<IntlApplication messages={messages} orgId={options.orgUnit.OrgId} />, parent);
```
And to access translated terms:
```
class SomeComponent extends React.Component {
	render() {
		console.log(this.context.getIntlMessage('MyApp.Save'));
                ...blah blah blah
	}
}
SomeComponent.contextTypes = {
	getIntlMessage: React.PropTypes.func
};
export default SomeComponent;
```

It seems likely (based on discussion) that `getIntlMessage` will eventually be deprecated from react-intl in favor of formatting components such as `FormatMessage`, `FormatDate`, etc., and are also likely to change.  Defining formatting components eliminates the need for the app to interact directly with `getIntlMessage`, and therefore also eliminates the need for the app components to be aware of  `contextTypes`.

After much head banging on desk, I am not completely happy, but I don't think I will be until React goes to 0.14 and react-intl is updated.

### CODE REVIEW

**Comment by dlockhart** [comment]
So if instead of using mixins, we just went the way of 100% using formatting components like `FormatMessage`, would any of this stuff be necessary? Or would we still need to wrap it so that those components could somehow know what language/locale/etc to use?

File: package.json [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
   "description": "A simple React mixin for getting localized text in components.",
-  "main": "src/react-frau-intl.js",
+  "main": "src/index.js",
   "scripts": {
     "lint": "npm run lint:src && npm run lint:test",
     "lint:src": "jshint src",
     "lint:test": "jshint test",
     "test:unit": "istanbul cover ./node_modules/jasmine-node/bin/jasmine-node -- test --matchall",
-    "test": "npm run lint && npm run test:unit",
+    "test": "npm run test:unit",
```
Comment (dlockhart) [inline package.json]: FYI, if you want to keep the linting goodness, there's an `esnext` flag you can use

Comment (dbatiste) [inline package.json]: Cool yeah - was wondering about that.  :)

**Comment by dbatiste** [comment]
> So if instead of using mixins, we just went the way of 100% using formatting components like FormatMessage, would any of this stuff be necessary? Or would we still need to wrap it so that those components could somehow know what language/locale/etc to use?

Yeah - the formatting components such as `FormatMessage` would still need access to the language/locale data, and in this case, that would be provided by context and is initialized at the root of their app.  In the case of formatting components, consumers would not need to reference `context`, `getIntlMessage` directly, or set up `contextTypes` since this would be done inside the formatting component(s).

It is worth noting that the factory method `internationalize` is only needed because React context currently follows the ownership chain (rather than the parent-child), however this behavior is changing with React 0.14.  At that point in time, the factory method will no longer be required, and initializing at the root will look like:

```
   <Intl messages={messages} ... >
      <Application />
   </Intl>
```

Another thing worth mentioning is that this does not rely on browser supporting Intl, or any of the locale data provided by react-intl.  I think that locale data is used to support date, time, number, and other formatting as per CLDR formats for the various locales.  Do we care if we don't have that?  Our formats will be coming from elsewhere correct?  I suspect eventually we will want to adopt the refactored react-intl when it's ready in order to get the benefits of some of the awesomeness (such as plurality) that's possible with the ICU formatting, but it probably wouldn't be dates, times, numbers, or currency that we care about.

**Comment by dlockhart** [comment]
Cool! Yeah I think we're going to write our own formatters for dates, times, numbers, etc. since all that stuff is very configurable and isn't necessarily tied to the locale itself. Instead it's tied to a bunch of settings within Brightspace -- some of which are overridable by the user.

**Pull request merged**